pusher/src/Controller/IoSocketController.ts

```diff
@@ -694,7 +695,7 @@ export class IoSocketController {
         client.token = ws.token;
         client.batchedMessages = new BatchMessage();
         client.batchTimeout = null;
-        client.emitInBatch = (payload: SubMessage): void => {
+        client.emitInBatch = (payload: SubMessage | SubChatMessage): void => {
```

Comment:
The whole initClient method is just plain wrong.

Idea:

Out of initClient, add emitInBatch or emitInBatchChat

Validate the client with Zod (with 2 zod types): ExSocketInterface and ExChatSocketInterface